library/std/src/io/mod.rs

```diff
@@ -386,20 +398,12 @@ where
         }
 
         match r.read(&mut g.buf[g.len..]) {
-            Ok(0) => {
-                ret = Ok(g.len - start_len);
-                break;
-            }
+            Ok(0) => return Ok(g.len - start_len),
```

Comment:
This bit down here is just a drive-by cleanup to some weird control flow left after the commit mentioned in the PR description.